quickwit/quickwit-doc-mapper/src/default_doc_mapper/date_time_type.rs

```diff
@@ -135,6 +138,47 @@ mod tests {
 
     #[test]
     fn test_date_time_options_single_value_deser() {
+        let field_mapping_entry = serde_json::from_str::<FieldMappingEntry>(
+            r#"
+            {
+                "name": "updated_at",
+                "type": "datetime",
+                "description": "When the record was last updated.",
+                "input_formats": [
+                    "rfc3339"
+                ],
+                "fast_precision": "milliseconds",
+                "indexed": true,
+                "fast": true,
+                "stored": false
+            }
+            "#,
+        )
+        .unwrap();
+
+        assert_eq!(field_mapping_entry.name, "updated_at");
+
+        let date_time_options = match field_mapping_entry.mapping_type {
+            FieldMappingType::DateTime(date_time_options, Cardinality::SingleValue) => {
+                date_time_options
+            }
+            _ => panic!("Expected a date time field mapping."),
+        };
+        let expected_input_formats = InputFormats(vec![DateTimeInputFormat::Rfc3339]);
+        let expected_date_time_options = QuickwitDateTimeOptions {
+            description: Some("When the record was last updated.".to_string()),
+            input_formats: expected_input_formats,
+            output_format: DateTimeOutputFormat::Rfc3339,
+            fast_precision: DateTimePrecision::Milliseconds,
+            indexed: true,
+            fast: true,
+            stored: false,
+        };
+        assert_eq!(date_time_options, expected_date_time_options);
+    }
+
+    #[test]
+    fn test_date_time_options_single_value_deser_backward_compatible() {
```

Comment:
All this backward compatibility test needs to do is deserialize the datetime option and check that the `fast_precison` field was populated correctly. We don't need the remaining duplicate code/test.